From ralf@linux-mips.org Sat Jun  3 15:59:05 2006
Date: Sat, 3 Jun 2006 23:58:55 +0100
From: Ralf Baechle <ralf@linux-mips.org>
To: gregkh@suse.de, dbrownell@users.sourceforge.net,
Subject: USB: EHCI on non-Au1200 build fix
Message-ID: <20060603225855.GA2234@linux-mips.org>
Content-Disposition: inline

From: Ralf Baechle <ralf@linux-mips.org>

Including ehci-au1xxx.c on a non-Au1200 Alchemy only to have it throw
an error is stupid.

From: Ralf Baechle <ralf@linux-mips.org>
Acked-by: David Brownell <david-b@pacbell.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


---
 drivers/usb/host/ehci-au1xxx.c |    4 ----
 drivers/usb/host/ehci-hcd.c    |    2 +-
 2 files changed, 1 insertion(+), 5 deletions(-)

--- gregkh-2.6.orig/drivers/usb/host/ehci-au1xxx.c
+++ gregkh-2.6/drivers/usb/host/ehci-au1xxx.c
@@ -16,10 +16,6 @@
 #include <linux/platform_device.h>
 #include <asm/mach-au1x00/au1000.h>
 
-#ifndef CONFIG_SOC_AU1200
-#error "this Alchemy chip doesn't have EHCI"
-#else				/* Au1200 */
-
 #define USB_HOST_CONFIG   (USB_MSR_BASE + USB_MSR_MCFG)
 #define USB_MCFG_PFEN     (1<<31)
 #define USB_MCFG_RDCOMB   (1<<30)
--- gregkh-2.6.orig/drivers/usb/host/ehci-hcd.c
+++ gregkh-2.6/drivers/usb/host/ehci-hcd.c
@@ -897,7 +897,7 @@ MODULE_LICENSE ("GPL");
 #define	PLATFORM_DRIVER		ehci_fsl_driver
 #endif
 
-#ifdef CONFIG_SOC_AU1X00
+#ifdef CONFIG_SOC_AU1200
 #include "ehci-au1xxx.c"
 #define	PLATFORM_DRIVER		ehci_hcd_au1xxx_driver
 #endif
